commands/http/handler.go

```diff
@@ -422,3 +426,21 @@ func allowReferer(r *http.Request, cfg *ServerConfig) bool {
 
 	return false
 }
+
+// apiVersionMatches checks whether the api server is running the
+// same version of go-ipfs. for now, only the exact same version of
+// client + server work. In the future, we should use semver for
+// proper API versioning! \o/
+func apiVersionMatches(r *http.Request) error {
+	clientVersion := r.UserAgent()
+	// skips check if client is not go-ipfs
+	if clientVersion == "" || !strings.Contains(clientVersion, "/go-ipfs/") {
+		return nil
+	}
```

Comment:
maybe we could use a dedicated header to ask for a version match, something like:

```yml
X-IPFS-Match-Version: go-ipfs/0.3.9
# or
X-IPFS-Match-API-Version: ipfs-api/0.1.0
```
it's tricky because we'll have clients and so on. maybe what we should be doing here is starting to define the API itself with `semver`, like: `ipfs-api/0.1.2` and matching on _that_. (the previous check was meant as a temporary thing anyway)

cc @dignifiedquire @diasdavid